src/power.jl

```diff
@@ -40,6 +40,34 @@ for T in (:Taylor1, :TaylorN)
 end
 
 
+# power_by_squaring; slightly modified from base/intfuncs.jl
```

Comment:
Previously, in  487cb8b, we had this function, which turns out that it improves the performance with respect to using `Base.power_by_squaring`. 

The reason is that, wherever you read `square(x)`, in Base it is written as `x*x`, and `square(x)` is faster and uses less memory.